### PR TITLE
Converter draft

### PR DESCRIPTION
**Pull request by VedPatwardhan**

### CODE REVIEW

**Comment by YushaArif99** [comment]
import haiku as hk
import jax
import jax.numpy as jnp
import ivy


class HaikuLinear(hk.Module):
  def __init__(self, out_size):
    super(HaikuLinear, self).__init__()
    self._linear = hk.Linear(out_size)

  def __call__(self, x):
    return self._linear(x)


class HaikuModule(hk.Module):
  def __init__(self, out_size, device=None, hidden_size=64):
    super(HaikuModule, self).__init__()
    self._linear0 = HaikuLinear(hidden_size)
    self._linear1 = HaikuLinear(hidden_size)
    self._linear2 = HaikuLinear(out_size)

  def __call__(self, x):
    x = jnp.expand_dims(x, 0)
    x = jnp.tanh(self._linear0(x))
    x = jnp.tanh(self._linear1(x))
    return jnp.tanh(self._linear2(x))[0]


def forward_fn(*a, **kw):
  model = HaikuModule(10)
  return model(*a, **kw)


ivy.set_backend('jax')
batch_shape, input_channels, output_channels = [1,2], 4, 5
x = ivy.astype(
        ivy.linspace(ivy.zeros(batch_shape), ivy.ones(batch_shape), input_channels),
        "float32",
    )


def forward_fn(*a, **kw):
  model = HaikuModule(input_channels, output_channels)
  return model(*a, **kw)

def loss_fn(v_=None):
  out = ivy_module(x, v=v_)
  return ivy.mean(out)


# train
loss_tm1 = 1e12
loss = None
grads = None

native_module = hk.transform(forward_fn)
ivy_module = ivy.to_ivy_module(
  native_module, inplace_update=False,
    args=[x]
)
for i in range(100):
    loss, grads = ivy.execute_with_gradients(loss_fn, ivy_module.v)
    ivy_module.v = ivy.gradient_descent_update(ivy_module.v, grads, 1e-3)
    assert loss < loss_tm1
    loss_tm1 = loss
    print(loss_tm1)

**Comment by djl11** [comment]
@YushaArif99 please continue with this task, using this PR as a reference :)
